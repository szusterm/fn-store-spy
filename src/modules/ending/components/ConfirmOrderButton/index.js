import ConfirmOrderButton from './ConfirmOrderButton';

export default ConfirmOrderButton;