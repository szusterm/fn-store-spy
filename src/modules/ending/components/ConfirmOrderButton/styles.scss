@import '../../../../../node_modules/material-colors/dist/colors';

@import '../../../../scss/shadow';
@import '../../../../scss/transition';
@import '../../../../scss/input';
@import '../../../../scss/opacity-colors';
@import '../../../../scss/variables';

.confirm-order-button {
	@include default-input();

	background-color: $md-blue-500;
	font-size: 1.3em;

	@include white-color(0.9);
	@include shadow(sm);
	@include transition(sh);

	&:hover, &:focus {
		background-color: $md-blue-600;
		@include shadow(md);
	}

	&:active {
		background-color: $md-blue-700;
	}

	&:disabled {
		cursor: default;
		@include black-background(0.1);
		@include black-color(0.5);

		&:hover, &:focus {
			@include shadow(sm);
		}
	}
}