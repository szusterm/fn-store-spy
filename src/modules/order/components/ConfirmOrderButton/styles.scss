@import '~material-colors/dist/colors';

@import '~scss/_shadow';
@import '~scss/_transition';
@import '~scss/_input';
@import '~scss/_opacity-colors';
@import '~scss/_variables';

.confirm-order-button {
	@include default-input();

	background-color: $md-blue-500;
	font-size: 1.3em;

	@include white-color(0.9);
	@include shadow(sm);
	@include transition(sh);

	&:hover, &:focus {
		background-color: $md-blue-600;
		@include shadow(md);
	}

	&:active {
		background-color: $md-blue-700;
	}
}