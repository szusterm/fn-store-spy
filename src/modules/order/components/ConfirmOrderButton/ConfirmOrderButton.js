import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {sendOrder} from '../../actions';

import './styles.scss';

export class ConfirmOrderButton extends Component {
	constructor(props) {
		super(props);

		this.state = {
			sendingOrder: false
		};

		this.sendOrder = this.sendOrder.bind(this);
	}

	async sendOrder() {
		this.setSendingOrderState(true);

		const itemsIds = this.getItemsIds();
		await this.props.sendOrder(itemsIds);

		this.setSendingOrderState(false);
	}

	getItemsIds() {
		const {orderedItems} = this.props;

		const itemsIds = [];
		for (const {_id: orderedItemId} of orderedItems) {
			itemsIds.push(orderedItemId);
		}

		return itemsIds;
	}

	setSendingOrderState(sendingOrder) {
		const newState = Object.assign(
			this.state,
			{
				sendingOrder
			}
		);

		this.setState(newState);
	}

	checkThatOrderIsEmpty() {
		const {orderedItems} = this.props;

		return (orderedItems.length === 0);
	}

	checkThatOrderIsTooBig() {
		const {orderedItems, maxItemsInOrder} = this.props;

		return (orderedItems.length > maxItemsInOrder);
	}

	render() {
		const {sendingOrder} = this.state;

		return (
			<button
				className={'confirm-order-button'}
				disabled={(this.checkThatOrderIsEmpty() || this.checkThatOrderIsTooBig() || sendingOrder)}
				onClick={this.sendOrder}
			>
				Send order
			</button>
		);
	}
}

ConfirmOrderButton.propTypes = {
	orderedItems: PropTypes.array, //redux
	maxItemsInOrder: PropTypes.number, //redux
	sendOrder: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {items} = state.order;
	const {maxItemsInOrder} = state.config;

	return {
		orderedItems: items,
		maxItemsInOrder
	};
};

const mapDispatchToProps = {sendOrder};

export default connect(mapStateToProps, mapDispatchToProps)(ConfirmOrderButton);