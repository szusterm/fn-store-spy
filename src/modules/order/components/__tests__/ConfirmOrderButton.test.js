import React from 'react';
import {shallow} from 'enzyme';

import {ConfirmOrderButton} from '../ConfirmOrderButton/ConfirmOrderButton';

const setup = (propOverrides) => {
	const props = Object.assign({
		orderedItems: [], //redux
		maxItemsInOrder: 3 //redux
	}, propOverrides);

	const wrapper = shallow(<ConfirmOrderButton {...props}/>);

	return {props, wrapper};
};

describe('CircleOrderCount Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});
});