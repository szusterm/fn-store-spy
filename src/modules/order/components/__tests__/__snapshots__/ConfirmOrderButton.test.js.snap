// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmOrderButton Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmOrderButton
    maxItemsInOrder={3}
    orderedItems={
      Array [
        "item1",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Send order",
      "className": "confirm-order-button",
      "disabled": false,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Send order",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Send order",
        "className": "confirm-order-button",
        "disabled": false,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Send order",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
